test/integration/MonadicSystemSpec.hs

```diff
@@ -0,0 +1,166 @@
+module MonadicSystemSpec (spec) where
+
+    import Test.QuickCheck
+    import Test.Hspec
+    import Lexer
+    import Parser
+    import Eval
+    import qualified Data.Map as Map
+
+    -- Test suites --
+    spec :: Spec
+    spec = do
+        tsFunctions
+
+
+    -- Linking tests to Test Suites --
+
+    tsFunctions = describe "functions" $ do
+        testT
```

Comment:
Maybe make the test names more declarative, so we know what tests what.

Reply:
Good idea, I'll fix it now!